### PR TITLE
service: nfc: Update Implementation to match with latest RE

### PR DESCRIPTION
**Pull request by german77**
Fixes some minor nits and implements ListDevices, AttachActivateEvent, AttachDeactivateEvent closer to my RE.

### CODE REVIEW

File: src/core/hle/service/nfc/common/device_manager.cpp [hunk 1]
```diff
@@ -51,22 +52,52 @@ Result DeviceManager::Finalize() {
...
+    }
+
     for (auto& device : devices) {
         if (nfp_devices.size() >= max_allowed_devices) {
             continue;
         }
-        if (device->GetCurrentState() != DeviceState::Unavailable) {
-            nfp_devices.push_back(device->GetHandle());
+        if (skip_fatal_errors) {
+            constexpr u64 MinimuRecoveryTime = 60;
```
Comment (liamwhite) [inline src/core/hle/service/nfc/common/device_manager.cpp:81]:
```suggestion
            constexpr u64 MinimumRecoveryTime = 60;
```

File: src/core/hle/service/nfc/common/device_manager.cpp [hunk 2]
```diff
@@ -142,24 +173,46 @@ Result DeviceManager::GetTagInfo(u64 device_handle, TagInfo& tag_info) const {
     return result;
 }
 
-Kernel::KReadableEvent& DeviceManager::AttachActivateEvent(u64 device_handle) const {
-    std::scoped_lock lock{mutex};
```
Comment (liamwhite) [inline src/core/hle/service/nfc/common/device_manager.cpp:146]: No longer requires the lock?

Comment (german77) [reply src/core/hle/service/nfc/common/device_manager.cpp:146]: The sysmodule doesn't have it here. Instead is on listDevices

Comment (liamwhite) [reply src/core/hle/service/nfc/common/device_manager.cpp:146]: Should you be taking it in ListDevices, then?

Comment (german77) [reply src/core/hle/service/nfc/common/device_manager.cpp:146]: Yeah. I forgot that one now that I check the code. Thanks.

File: src/core/hle/service/nfc/common/device_manager.cpp [hunk 3]
```diff
@@ -593,6 +647,20 @@ Result DeviceManager::WriteNtf(u64 device_handle, NFP::WriteType, std::span<cons
     return result;
 }
 
+Result DeviceManager::CheckHandleOnList(u64 device_handle, std::span<u64> device_list) const {
+    if (device_list.size() < 1) {
+        return ResultDeviceNotFound;
+    }
+
+    for (u64 handle : device_list) {
```
Comment (FearlessTobi) [inline src/core/hle/service/nfc/common/device_manager.cpp:655]: if device_list were a vector this all could just be a call to `contains`

Comment (v1993) [reply src/core/hle/service/nfc/common/device_manager.cpp:655]: Why not keep it as span and do

```cpp
std::ranges::contains(device_list, device_handle);
```

Comment (german77) [reply src/core/hle/service/nfc/common/device_manager.cpp:655]: std::ranges::contains is c++23 we are at c++22

Comment (german77) [reply src/core/hle/service/nfc/common/device_manager.cpp]: I replaced the for loop with sdt::find. Doesn't make any difference for me but works as well.

**Review by liamwhite — APPROVED** [review]

**Pull request merged**